src/services/spot/Spot.cpp

```diff
@@ -0,0 +1,387 @@
+            }
+
+            s_instance.reset(new Spot());
+        }
+
+        static void write_snapshot_cb(Caliper* c, const SnapshotRecord*, const SnapshotRecord* snapshot) {
+            if (!s_instance)
+                return;
+
+            s_instance->process_snapshot(c, snapshot);
+        }
+
+        static void post_write_cb(Caliper* c, const SnapshotRecord* flush_info) {
+            if (!s_instance)
+                return;
+
+            s_instance->flush(c, flush_info);
+        }
+
+    public:
+
+        ~Spot()
+            { }
+
+        static void create(Caliper* c) {
+            c->events().pre_write_evt.connect(pre_write_cb);
+            c->events().write_snapshot.connect(write_snapshot_cb);
+            c->events().post_write_evt.connect(post_write_cb);
+
+            Log(1).stream() << "Registered Spot service" << std::endl;
+        }
+    };
+
+    std::unique_ptr<Spot> Spot::s_instance { nullptr };
+    int Spot::divisor = 1000000;
+    Spot::List<std::pair<Spot::AggregationHandler,Spot::SelectionHandler>> Spot::m_queries;
+    Spot::JsonListType Spot::m_jsons;
+    Spot::AggregationDescriptorList Spot::m_annotations_and_places; 
+    std::vector<std::string> Spot::y_axes;
+    std::string Spot::code_version;
+    std::string Spot::recorded_time;
+    std::vector<std::string> Spot::title;
+    bool Spot::include_local;
+    bool Spot::show_exclusive;
+    const ConfigSet::Entry  Spot::s_configdata[] = {
+        { "config", CALI_TYPE_STRING, "function:default.json",
+          "Attribute:Filename pairs in which to dump Spot data",
+          "Attribute:Filename pairs in which to dump Spot data\n"
+          "Example: function:testname.json,physics_package:packages.json"
+          "   stderr: Standard error stream,\n"
+          " or a file name.\n"
+        },
+        { "recorded_time", CALI_TYPE_STRING, "",
+          "Time to use for this version of the code",
+          "Time to use for this version of the code"
+        },
+        { "code_version", CALI_TYPE_STRING, "unspecified",
+          "Version number (or git hash) to represent this run of the code",
+          "Version number (or git hash) to represent this run of the code"
+        },
+        { "time_divisor", CALI_TYPE_INT, "1000000",
+          "Caliper records time in microseconds, this is what we divide by to get time in your units",
+          "Caliper records time in microseconds, this is what we divide by to get time in your units. 1000 if you record in milliseconds, 1000000 if seconds"
+        },
+        { "y_axes", CALI_TYPE_INT, "microseconds",
+          "If this is the first time Spot has seen a test, tell it what Y Axis to display on the resulting graphs. If multiple graphs, separate entries with a colon (:)",
+          "If this is the first time Spot has seen a test, tell it what Y Axis to display on the resulting graphs. If multiple graphs, separate entries with a colon (:)"},
+        { "include_local", CALI_TYPE_STRING, "YES",
```

Comment:
Why not make this booleans instead of strings? Caliper should handle 0, 1, on, off then. We can add yes and no, too.